src/test/java/net/snowflake/client/jdbc/HeartbeatAsyncLatestIT.java

```diff
@@ -43,20 +43,32 @@ protected void submitQuery(boolean useKeepAliveSession, int queryIdx)
       connection = getConnection(sessionParams);
 
       Statement stmt = connection.createStatement();
-      // Query will take 30 seconds to run, but ResultSet will be returned immediately
+      // Query will take 5 seconds to run, but ResultSet will be returned immediately
       resultSet =
           stmt.unwrap(SnowflakeStatement.class)
-              .executeAsyncQuery("SELECT count(*) FROM TABLE(generator(timeLimit => 30))");
+              .executeAsyncQuery("SELECT count(*) FROM TABLE(generator(timeLimit => 5))");
       Thread.sleep(61000); // sleep 61 seconds to await original session expiration time
       QueryStatus qs = resultSet.unwrap(SnowflakeResultSet.class).getStatus();
-      // Ensure query succeeded
-      assertEquals(QueryStatus.SUCCESS, qs);
+      int retry = 0;
+      int MAX_RETRY = 10;
+      // Ensure query succeeded. Avoid flaky test failure by waiting until query is complete to
+      // assert the query status is a success.
+      while (QueryStatus.isStillRunning(qs) && retry < MAX_RETRY) {
+        Thread.sleep(1000);
+        retry++;
+        qs = resultSet.unwrap(SnowflakeResultSet.class).getStatus();
+      }
+      if (retry < MAX_RETRY) {
+        assertEquals(QueryStatus.SUCCESS, qs);
 
-      // assert we get 1 row
-      assertTrue(resultSet.next());
-      assertFalse(resultSet.next());
+        // assert we get 1 row
+        assertTrue(resultSet.next());
+        assertFalse(resultSet.next());
+        logger.fine("Query " + queryIdx + " passed ");
+      } else {
+        logger.fine("Query took too long to complete. Aborting test.");
```

Comment:
should we report the issue instead of silence it?

Reply:
Maybe we can keep the assertion here, considered we already change the run time to 5s.

Reply:
This is true- I increased the retry time and added the assertion back in.